src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -247,4 +258,33 @@ public boolean verify(String hostname, SSLSession session) {
     };
     HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
   }
+
+  private void flushMessageBuffer() {
+    int batchSize = config.getBatchSize();
+    String jsonBatchKey = config.getJsonBatchKey();
+    boolean useJsonBatchKey = !Strings.isNullOrEmpty(jsonBatchKey) && batchSize > 1;
+    boolean shouldWriteJsonAsArray = config.shouldWriteJsonAsArray();
+    switch (config.getMessageFormat()) {
+      case "JSON":
+        String messageBody;
+        if (!shouldWriteJsonAsArray) {
```

Comment:
these checks would be easier to understand if structured as:

```
if (!shouldWriteJsonAsArray) {
  ...
  break;
}
if (useJsonBatchKey) {
  ...
} else {
  ...
}
```

Reply:
Updated
```java
if (!config.shouldWriteJsonAsArray()) {
  // ...
  break;
}
if (useJsonBatchKey) {
 // ...
} else {
  ...
}
```